src/web3/index.ts

```diff
@@ -31,8 +41,23 @@ export const CHAINS: {
 };
 
 export function getTriggerClass(
-  params: ConnectorInput<{ chain: string | string[] }>
-): new (params: ConnectorInput) => TriggerBase {
+  params: TriggerInit<NewEventInput, TriggerBasePayload, TriggerBaseState>
+): TriggerBaseEventConstructor;
+
+export function getTriggerClass(
+  params: TriggerInit<NewTransactionInput, TriggerBasePayload, TriggerBaseState>
+): TriggerBaseTxConstructor;
+
+export function getTriggerClass(
+  params: TriggerInit<NewTransactionFlowInput, TriggerBasePayload, TriggerBaseState>
+): TriggerBaseTxFlowConstructor;
```

Comment:
@SAPikachu On this part this is more tricky because in `setupSignal` there is the `return new Trigger(params);` line that is related to the result of `getTriggerClass` function. Always having type error here if removing the `getTriggerClass` overload. 

I imagine that there is a solution by typing but it could be studied on another PR, right? I just wanted to go simple here.

Reply:
@SAPikachu I've just removed the overloads with a better conditional typing, especially for the line
```typescript
return new Trigger(params as TriggerBaseConstructor<typeof Trigger>);
```
in the `setupSignal` function which was sensitive to this typing issue. Tell me if it fits your expectations and don't hesitate to modify if needed.